### PR TITLE
Minor S3 write refactor to try and fix reset error

### PR DESCRIPTION
**Pull request by theefer**
Attempts to fix the following error, which seems to occur when there is some network issue that requires the client to re-read data from the InputStream we pass to S3 putObject. Because it's not buffered I suspect it fails there. Using a different request constructor may help because it then uses a `ResettableInputStream` instead of the `FileInputStream` we pass it.

```
Caused by: com.amazonaws.ResetException: Failed to reset the request input stream;  If the request involves an input stream, the maximum stream buffer size can be configured via request.getRequestClientOptions().setReadLimit(int)
	at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:637) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:461) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:296) ~[app.jar:0.1]
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:3735) ~[app.jar:0.1]
	at com.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1460) ~[app.jar:0.1]
	at com.gu.mediaservice.lib.aws.S3$$anonfun$store$1.apply(S3.scala:49) ~[app.jar:0.1]
	at com.gu.mediaservice.lib.aws.S3$$anonfun$store$1.apply(S3.scala:41) ~[app.jar:0.1]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[app.jar:0.1]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[app.jar:0.1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_45-internal]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_45-internal]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_45-internal]
Caused by: java.io.IOException: Resetting to invalid mark
	at java.io.BufferedInputStream.reset(BufferedInputStream.java:448) ~[na:1.8.0_45-internal]
	at com.amazonaws.internal.SdkBufferedInputStream.reset(SdkBufferedInputStream.java:106) ~[app.jar:0.1]
	at com.amazonaws.internal.SdkFilterInputStream.reset(SdkFilterInputStream.java:103) ~[app.jar:0.1]
	at com.amazonaws.event.ProgressInputStream.reset(ProgressInputStream.java:139) ~[app.jar:0.1]
	at com.amazonaws.internal.SdkFilterInputStream.reset(SdkFilterInputStream.java:103) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:635) ~[app.jar:0.1]
```

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/lib/aws/S3.scala [hunk 1]
```diff
@@ -43,9 +43,8 @@ class S3(credentials: AWSCredentials) {
       mimeType.foreach(metadata.setContentType)
       cacheControl.foreach(metadata.setCacheControl)
       metadata.setUserMetadata(meta.asJava)
-      metadata.setContentLength(file.length)
```
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/aws/S3.scala]: Note that this is now done internally AFAICS.

**Comment by theefer** [comment]
I still need to do some more testing around this. Annoyingly hard to replicate the failure conditions but I want to at least prove this works as well as previously.

**Comment by theefer** [comment]
Works on TEST, I can't reproduce the error but it doesn't seem to degrade the cropping to I'd like to roll this out for testing.

Can someone review please?

**Comment by akash1810** [comment]
Are there any consequences to not using a FileStream?

**Comment by theefer** [comment]
Not that I know of, from the AWS SDK source it's just using a "better" file input stream internally instead.

**Comment by akash1810** [comment]
:+1:

**Comment by theefer** [comment]
Bit scared to do this so late in the day so will ship tomorrow morning.

**Pull request merged**